File: DSA-Python/5-array/8-linear-search.py
# Given an array, and an element num the task is to find if num is present in the given array or not. If present print the index of the element or print -1.



# O(n), O(1)
def linear_search(nums:list[int], target:int) -> bool:
    
    for x in nums:
        if (x == target):
            return True
        
    return False


if __name__ == "__main__":
    loops = int(input())

    for _ in range(loops):
        target = int(input())
        
        nums = list(map(int, input().split()))
        
        res = linear_search(nums, target)
        
        print(res)
        
        
        
'''
5
3
1 2 3 4 5
5
5 4 3 2 1
-3
0 0 -1 -3 -2 0 
-1
0
-1
'''

File: DSA-C++/6-array/8-linear-search.cpp
#include <bits/stdc++.h>

using namespace std;
// Given an array, and an element num the task is to find if num is present in the given array or not. If present print the index of the element or print -1.


// O(n), O(1)
int linear_search(vector <int> &nums, int target) {
    int n = nums.size();

    for (int i=0; i<n; ++i) {
        if (nums[i] == target) {
            return i;
        }
    }

    return -1;
}

int main() {
    vector <int> nums = {5,4,3,2,1};
    int target = 5;
    target = -5;

    cout << linear_search(nums, target) << endl;

    return 0;
}